cpp/src/lance/format/schema.cc

```diff
@@ -382,6 +382,48 @@ std::shared_ptr<Field> Field::Project(const std::shared_ptr<::arrow::Field>& arr
   return new_field;
 }
 
+::arrow::Result<std::shared_ptr<Field>> Field::Merge(const Field& other) const {
+  if (name_ != other.name_) {
+    return ::arrow::Status::Invalid(
+        fmt::format("Field::merge: attempt to merge two fields with different names: {} != {}",
+                    name_,
+                    other.name_));
+  };
+  if (logical_type_ != other.logical_type_) {
```

Comment:
some arrow native dtypes are parametric (e.g., timestamps). If say two timestamp fields have the same name but different basis, it would succeed here. Is this expected?

Reply:
changed to directly check type equallity.